cilium-cli/cli/clustermesh.go

```diff
@@ -293,6 +308,8 @@ func newCmdClusterMeshConnectWithHelm() *cobra.Command {
 		Long:  ``,
 		RunE: func(_ *cobra.Command, _ []string) error {
 			params.Namespace = namespace
+			params.ImpersonateAs = impersonateAs
+			params.ImpersonateGroups = impersonateGroups
```

Comment:
These are technically needed only for the `ConnectWithHelm` and `DisconnectWithHelm` commands, which are the only ones creating clients targeting remote clusters as well. But it doesn't hurt configuring them in the other cases as well.